console/src/main/java/org/matrix/console/UnrecognizedCertHandler.java

```diff
@@ -0,0 +1,159 @@
+package org.matrix.console;
+
+
+import android.app.Activity;
+import android.content.DialogInterface;
+import android.support.v7.app.AlertDialog;
+import android.view.LayoutInflater;
+import android.util.Log;
+import android.view.View;
+import android.widget.TextView;
+
+import org.matrix.androidsdk.HomeserverConnectionConfig;
+import org.matrix.androidsdk.MXSession;
+import org.matrix.androidsdk.ssl.Fingerprint;
+import org.matrix.console.activity.CommonActivityUtils;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class UnrecognizedCertHandler {
+    private static final String LOG_TAG = "UnrecognizedCertHandler";
+
+    private static HashMap<String, HashSet<Fingerprint>> ignoredFingerprints = new HashMap<String, HashSet<Fingerprint>>();
+    private static HashSet<String> openDialogs = new HashSet<String>();
```

Comment:
Nit: These aren't `Dialog` instances, but IDs, so this should probably be `openDialogIds` for clarity.